src/request/change-passphrase/ChangePassphraseApi.js

```diff
@@ -27,40 +26,20 @@ class ChangePassphraseApi extends TopLevelApi { // eslint-disable-line no-unused
     }
 
     /**
+     *
      * @param {KeyguardRequest.SimpleRequest} request
      * @returns {Promise<KeyguardRequest.ParsedSimpleRequest>}
-     * @private
      */
-    static async _parseRequest(request) {
+    async parseRequest(request) {
         if (!request) {
-            throw new Errors.InvalidRequestError('Request can not be empty');
+            throw new Errors.InvalidRequestError('request is required');
         }
 
-        // Check that keyId is given.
-        if (!request.keyId || typeof request.keyId !== 'string') {
-            throw new Errors.InvalidRequestError('keyId is required');
-        }
-
-        // Check that appName is given
-        if (!request.appName || typeof request.appName !== 'string') {
-            throw new Errors.InvalidRequestError('appName is required');
-        }
-
-        // Check that key exists.
-        const keyInfo = await KeyStore.instance.getInfo(request.keyId);
-        if (!keyInfo) {
-            throw new Errors.KeyNotFoundError();
-        }
-
-        // Validate labels.
-        if (request.keyLabel !== undefined && (typeof request.keyLabel !== 'string' || request.keyLabel.length > 64)) {
-            throw new Errors.InvalidRequestError('Invalid keyLabel');
-        }
+        const parsedRequest = {};
+        parsedRequest.appName = this.parseAppName(request.appName);
+        parsedRequest.keyInfo = await this.parseKeyId(request.keyId);
+        parsedRequest.keyLabel = this.parseLabel(request.keyLabel);
 
-        return /** @type {ParsedSimpleRequest} */ {
-            appName: request.appName,
-            keyInfo,
-            keyLabel: request.keyLabel,
-        };
+        return parsedRequest;
```

Comment:
It would be possible. It would also prevent you from doing additional checks as can be seen in DeriveAddressApi.js line 38.
I went with doing it in the same way for each of the requests.